tests/integration/test_authd/test_authd_key_request_exec_path.py

```diff
@@ -0,0 +1,170 @@
+'''
+copyright: Copyright (C) 2015-2021, Wazuh Inc.
+
+           Created by Wazuh, Inc. <info@wazuh.com>.
+
+           This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+type: integration
+
+brief: These tests will check the different errors that may appear by modifying
+       the path of the configurable executable (exec_path).
+
+tier: 0
+
+modules:
+    - authd
+
+components:
+    - manager
+
+daemons:
+    - wazuh-authd
+
+os_platform:
+    - linux
+
+os_version:
+    - Arch Linux
+    - Amazon Linux 2
+    - Amazon Linux 1
+    - CentOS 8
+    - CentOS 7
+    - CentOS 6
+    - Ubuntu Focal
+    - Ubuntu Bionic
+    - Ubuntu Xenial
+    - Ubuntu Trusty
+    - Debian Buster
+    - Debian Stretch
+    - Debian Jessie
+    - Debian Wheezy
+    - Red Hat 8
+    - Red Hat 7
+    - Red Hat 6
+
+references:
+    - https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/auth.html
+    - https://documentation.wazuh.com/current/user-manual/registering/key-request.html
+
+tags:
+    - key_request
+'''
+import os
+import shutil
+
+import pytest
+from wazuh_testing.fim import generate_params
+from wazuh_testing.tools import LOG_FILE_PATH, WAZUH_PATH
+from wazuh_testing.tools.configuration import load_wazuh_configurations
+from wazuh_testing.tools.file import read_yaml
+from wazuh_testing.authd import validate_authd_logs, override_wazuh_conf
+
+# Marks
+
+pytestmark = [pytest.mark.linux, pytest.mark.tier(level=0), pytest.mark.server]
+
+# Configurations
+
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+fetch_keys_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'files')
+message_tests = read_yaml(os.path.join(test_data_path, 'test_key_request_exec_path.yaml'))
+configurations_path = os.path.join(test_data_path, 'wazuh_authd_configuration.yaml')
+local_internal_options = {'authd.debug': '2'}
+filename = "fetch_keys.py"
+
+shutil.copy(os.path.join(fetch_keys_path, filename), os.path.join("/tmp", filename))
+
+DEFAULT_EXEC_PATH = '/var/ossec/framework/python/bin/python3 /tmp/fetch_keys.py'
+conf_params = {'EXEC_PATH': []}
+
+for case in message_tests:
+    conf_params['EXEC_PATH'].append(case.get('EXEC_PATH', DEFAULT_EXEC_PATH))
+
+p, m = generate_params(extra_params=conf_params, modes=['scheduled'] * len(message_tests))
+configurations = load_wazuh_configurations(configurations_path, __name__, params=p, metadata=m)
```

Comment:
Using `get_configuration/configure_environment` with `override_wazuh_conf` could be confusing. I recommend using only `override_wazuh_conf`. The configuration should be placed in only one file along with test case messages.

Also, this configuration should be applied using a fixture instead of a function inside the logic of the test.